src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -1040,38 +1040,80 @@ public class Program
+                //         Console.WriteLine(o is P { Unreadable2 is 4 });
+                Diagnostic(ErrorCode.ERR_InaccessibleGetter, "Unreadable2").WithArguments("P.Unreadable2").WithLocation(11, 36),
+                // (12,36): error CS0122: 'P.Unreadable3' is inaccessible due to its protection level
+                //         Console.WriteLine(o is P { Unreadable3 is 4 });
+                Diagnostic(ErrorCode.ERR_BadAccess, "Unreadable3").WithArguments("P.Unreadable3").WithLocation(12, 36)
                 );
             var tree = compilation.SyntaxTrees.Single();
             var model = compilation.GetSemanticModel(tree);
-            var propPats = tree.GetRoot().DescendantNodes().OfType<SubPropertyPatternSyntax>().ToArray();
-            Assert.Equal(2, propPats.Length);
 
-            var p = propPats[0]; // Length is 4
+            var propPats = tree.GetRoot().DescendantNodes().OfType<IsPatternExpressionSyntax>().Where(e => e.Parent is PropertyPatternSyntax).ToArray();
+            Assert.Equal(5, propPats.Length);
+
+            var p = propPats[0].Expression; // 'Good' in Good is 4
             var si = model.GetSymbolInfo(p);
             Assert.NotNull(si.Symbol);
-            Assert.Equal("Length", si.Symbol.Name);
+            Assert.Equal("Good", si.Symbol.Name);
             Assert.Equal(CandidateReason.None, si.CandidateReason);
             Assert.True(si.CandidateSymbols.IsDefaultOrEmpty);
 
-            p = propPats[1]; // NotFound is 4
+            p = propPats[1].Expression; // 'NotFound' in NotFound is 4
             si = model.GetSymbolInfo(p);
             Assert.Null(si.Symbol);
             Assert.Equal(CandidateReason.None, si.CandidateReason);
             Assert.True(si.CandidateSymbols.IsDefaultOrEmpty);
+
+            p = propPats[2].Expression; // 'Unreadable1' in Unreadable1 is 4
+            si = model.GetSymbolInfo(p);
+            Assert.Null(si.Symbol);
+            Assert.Equal(CandidateReason.NotAValue, si.CandidateReason);
+            Assert.Equal(1, si.CandidateSymbols.Length);
+            Assert.Equal("Unreadable1", si.CandidateSymbols[0].Name);
+
+            p = propPats[3].Expression; // 'Unreadable2' in Unreadable2 is 4
+            si = model.GetSymbolInfo(p);
+            Assert.Null(si.Symbol);
+            Assert.Equal(CandidateReason.NotAValue, si.CandidateReason);
+            Assert.Equal(1, si.CandidateSymbols.Length);
+            Assert.Equal("Unreadable2", si.CandidateSymbols[0].Name);
+
+            p = propPats[4].Expression; // 'Unreadable3' in Unreadable3 is 4
+            si = model.GetSymbolInfo(p);
+            Assert.Null(si.Symbol);
+            Assert.Equal(CandidateReason.Inaccessible, si.CandidateReason);
+            Assert.Equal(1, si.CandidateSymbols.Length);
+            Assert.Equal("Unreadable3", si.CandidateSymbols[0].Name);
         }
```

Comment:
Need to test events, methods, nested types. We should make sure the diagnostics given is reasonable and that compiler doesn't crash. It is Ok to refine diagnostics later, but the current behavior should be backed by a unit-test.